pkg/adapter/containers.go

```diff
@@ -694,3 +694,44 @@ func (r *LocalRuntime) UnpauseContainers(ctx context.Context, cli *cliconfig.Unp
 	}
 	return pool.Run()
 }
+
+// Prune removes stopped containers
+func (r *LocalRuntime) Prune(ctx context.Context, maxWorkers int, force bool) ([]string, map[string]error, error) {
+	var (
+		ok       = []string{}
+		failures = map[string]error{}
+		err      error
+	)
+
+	logrus.Debugf("Setting maximum rm workers to %d", maxWorkers)
+
+	filter := func(c *libpod.Container) bool {
+		state, err := c.State()
+		if state == libpod.ContainerStateStopped || (state == libpod.ContainerStateExited && err == nil && c.PodID() == "") {
```

Comment:
The logic here seems screwy.
Why are stopped and exited different? Why are we checking err (it's guaranteed that if err != nil, we never return a sensible state, only CtrStateUnknown)

Reply:
this is exactly what was in cmd/podman/containers_prune.go ... I changed nothing.  You'd have to ask the original author.

Reply:
I'm gonna propose we make the PodID check separate (if `podID != ""` always return false) and remove the err == nil check.